marklogic-client-api/src/main/java/com/marklogic/client/impl/TransformDefImpl.java

```diff
@@ -24,11 +26,36 @@ public StringBuilder exportAst(StringBuilder strb) {
         node.put("path", path);
         node.put("kind", kind);
         if (params != null) {
-            node.putPOJO("params", params);
+            node.putPOJO("params", prepareParamsDuringExport());
         }
         return strb.append(node);
     }
 
+    private Map<String, Object> prepareParamsDuringExport() {
```

Comment:
@anu3990  @llinggit  @josvanroosmalen  I couldn't find a better way to do this. The user's map can contain a variety of values - strings, numbers, booleans, ObjectNode's, and now PlanColumn's. So Jackson's `putPOJO` method is very convenient. But that ran into a stack overflow error on the `PlanColumn` class. So I worked around this by first serializing the `PlanColumn` into a JSON string and then converting that back into an `ObjectNode`. Everything works fine with this approach, just seems hacky.